zipkin-server/src/main/java/zipkin2/server/internal/elasticsearch/ZipkinElasticsearchStorageProperties.java

```diff
@@ -183,23 +138,32 @@ public void setHttpLogging(HttpLoggingLevel httpLogging) {
     this.httpLogging = httpLogging;
   }
 
-  public int getTimeout() {
+  public Integer getTimeout() {
     return timeout;
   }
 
-  public void setTimeout(int timeout) {
+  public void setTimeout(Integer timeout) {
     this.timeout = timeout;
   }
 
   public ElasticsearchStorage.Builder toBuilder() {
     ElasticsearchStorage.Builder builder = ElasticsearchStorage.newBuilder();
-    if (hosts != null) builder.hosts(hosts);
-    return builder
-        .index(index)
-        .dateSeparator(dateSeparator.isEmpty() ? 0 : dateSeparator.charAt(0))
-        .pipeline(pipeline)
-        .maxRequests(throttleMaxConcurrency == null ? maxRequests : throttleMaxConcurrency)
-        .indexShards(indexShards)
-        .indexReplicas(indexReplicas);
+    if (index != null) builder.index(index);
+    if (dateSeparator != null) {
+      builder.dateSeparator(dateSeparator.isEmpty() ? 0 : dateSeparator.charAt(0));
+    }
+    // TODO if (timeout != null) builder.timeout(timeout);
```

Comment:
This means having the storage receive timeout and setup the client appropriately I think - agree it would be cleaner than current where server passes in customizers to set the timeout.